clang/lib/CodeGen/CodeGenTypes.cpp

```diff
@@ -581,6 +581,9 @@ llvm::Type *CodeGenTypes::ConvertType(QualType T) {
   case BuiltinType::Id:                                                        \
     return llvm::TargetExtType::get(getLLVMContext(), "amdgcn.named.barrier",  \
                                     {}, {Scope});
+#define AMDGPU_FEATURE_PREDICATE_TYPE(Name, Id, SingletonId, Width, Align)     \
+  case BuiltinType::Id:                                                        \
+    return llvm::IntegerType::getInt1Ty(getLLVMContext());
```

Comment:
Why an int-1 type instead of 'bool' type?  Won't this cause problems if it is returned? Are we making sure we force casts correctly, else this is going to be a bug factory when emitting it.

Reply:
It was a leftover from an earlier iteration, thank you for flagging it, corrected. Having said that, the custom type should never get emitted in IR, the latest point where it'd still be present is at builtin CodeGen when targeting AMDGCN SPIR-V, and at that point it gets lowered into a load from a global bool. Someone could have gone and done e.g. a bitcast, I guess, but that's UB ultimately.